Won't read GOOGLE_APPLICATION_CREDENTIALS despite being present

I am trying to run `npx google-artifactregistry-auth` to publish to a GAR repo. Authentication, terraform and everything else pretty much works but the tool refuses to acknowledge `GOOGLE_APPLICATION_CREDENTIALS` despite it being set for the step.
These are the logs of the step from the GitHub workflow
```
Run npx google-artifactregistry-auth
  npx google-artifactregistry-auth
  shell: /usr/bin/bash -e {0}
  env:
    GOOGLE_APPLICATION_CREDENTIALS: /runner/_work/_temp/800a3c4b58ce99059c69334d
npm WARN exec The following package was not found and will be installed: google-artifactregistry-auth@3.1.0
Retrieving application default credentials...
Retrieving credentials from gcloud...
Error: Fail to get credentials. Please run: 
`gcloud auth application-default login`, `gcloud auth login`, or 
`export GOOGLE_APPLICATION_CREDENTIALS=<path/to/service/account/key>`
    at Object.getCreds (/home/runner/.npm/_npx/64aef35f3ba01c[7](https://github.sebank.se/hephaestus/verification/runs/964560?check_suite_focus=true#step:6:8)c/node_modules/google-artifactregistry-auth/src/auth.js:40:9)
    at async main (/home/runner/.npm/_npx/64aef35f3ba01c7c/node_modules/google-artifactregistry-auth/src/main.js:[8](https://github.sebank.se/hephaestus/verification/runs/964560?check_suite_focus=true#step:6:9)3:1[9](https://github.sebank.se/hephaestus/verification/runs/964560?check_suite_focus=true#step:6:10))
Error: Process completed with exit code 1.
```

So at first, without setting up gcloud before:
```
npm WARN exec The following package was not found and will be installed: google-artifactregistry-auth@3.1.0
Retrieving application default credentials...
Failed to retrieve application default credentials: client.getTokenInfo is not a function. Fall back to gcloud credentials.
Retrieving credentials from gcloud...
Failed to retrieve credentials from gcloud: Command failed: gcloud auth print-access-token
/bin/sh: 1: gcloud: not found
.
Error: Fail to get credentials. Please run: 
`gcloud auth application-default login`, `gcloud auth login`, or 
`export GOOGLE_APPLICATION_CREDENTIALS=<path/to/service/account/key>`
    at Object.getCreds (/home/runner/.npm/_npx/64aef35f3ba01c7c/node_modules/google-artifactregistry-auth/src/auth.js:40:9)
    at async main (/home/runner/.npm/_npx/64aef35f3ba01c7c/node_modules/google-artifactregistry-auth/src/main.js:83:19)
Error: Process completed with exit code 1.
```
With gcloud set up:
```
npm WARN exec The following package was not found and will be installed: google-artifactregistry-auth@3.1.0
Retrieving application default credentials...
Failed to retrieve application default credentials: Error code invalid_grant: The issuer in ID Token https://github.company.com/_services/token does not match the expected ones: https://spiffe.company.com.. Fall back to gcloud credentials.
Retrieving credentials from gcloud...
Failed to retrieve credentials from gcloud: Command failed: gcloud auth print-access-token
ERROR: (gcloud.auth.print-access-token) ('Error code invalid_grant: The issuer in ID Token https://github.company.com/_services/token does not match the expected ones: https://spiffe.company.com.', '{"error":"invalid_grant","error_description":"The issuer in ID Token https://github.company.com/_services/token does not match the expected ones: https://spiffe.company.com."}')
.
Error: Fail to get credentials. Please run: 
`gcloud auth application-default login`, `gcloud auth login`, or 
`export GOOGLE_APPLICATION_CREDENTIALS=<path/to/service/account/key>`
    at Object.getCreds (/home/runner/.npm/_npx/64aef35f3ba01c7c/node_modules/google-artifactregistry-auth/src/auth.js:40:9)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async main (/home/runner/.npm/_npx/64aef35f3ba01c7c/node_modules/google-artifactregistry-auth/src/main.js:83:19)
Error: Process completed with exit code 1.
```

Seeing the second error, it does make me think that something might be wrong with our company's set up of SPIFFE. I'll forward this internally to those who manage this.

--- a/src/auth.js
+++ b/src/auth.js
@@ -54,16 +54,7 @@ async function getApplicationDefaultCredentials() {
     projectId: 'unused-project'
   });
   const client = await auth.getClient();
-  const creds = (await client.getAccessToken()).token;
-  const tokenScopes = (await client.getTokenInfo(creds)).scopes;
-  if (!tokenScopes.includes(
-          'https://www.googleapis.com/auth/cloud-platform')) {
-    throw new Error(
-        'Token has insufficient authentication scopes.\n' +
-        'Please configure access scope following instructions on ' +
-        'https://cloud.google.com/artifact-registry/docs/access-control#compute');
-  }
-  return creds; 
+  return (await client.getAccessToken()).token;
 }
 
 /**

